test/functional/sqlsrv/sqlsrv_buffered.phpt

```diff
@@ -1,46 +1,82 @@
 --TEST--
 Read, Update, Insert from a SQLSRV stream with buffered query
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require('skipif_versions_old.inc'); ?>
 --FILE--
 <?php
 
 require_once('MsCommon.inc');
 
-$conn = connect();
-if (!$conn) {
-    fatalError("Could not connect");
-}
-
-$query = "IF OBJECT_ID('PhpCustomerTable', 'U') IS NOT NULL DROP TABLE [PhpCustomerTable]";
-$stmt = sqlsrv_prepare($conn, $query, array(), array("Scrollable"=>SQLSRV_CURSOR_CLIENT_BUFFERED));
-
-if (!$stmt) {
-    echo "Statement could not be prepared.\n";
-    die(print_r(sqlsrv_errors(), true));
-}
-
-sqlsrv_execute($stmt);
-
-$query = "CREATE TABLE [PhpCustomerTable] ([Id] int NOT NULL Identity (100,2) PRIMARY KEY, [Field2] text, [Field3] image, [Field4] ntext, [Field5] varbinary(max), [Field6] varchar(max), [Field7] nvarchar(max))";
-$stmt = sqlsrv_prepare($conn, $query, array(), array("Scrollable"=>SQLSRV_CURSOR_CLIENT_BUFFERED));
-
-if (!$stmt) {
-    echo "Statement could not be prepared.\n";
-    die(print_r(sqlsrv_errors(), true));
-}
-
-sqlsrv_execute($stmt);
-
-$query = "INSERT [PhpCustomerTable] ([Field2], [Field3], [Field4], [Field5], [Field6], [Field7]) VALUES ('This is field 2.', 0x010203, 'This is field 4.', 0x040506, 'This is field 6.', 'This is field 7.' )";
-$stmt = sqlsrv_prepare($conn, $query, array(), array("Scrollable"=>SQLSRV_CURSOR_CLIENT_BUFFERED));
-
-if (!$stmt) {
-    echo "Statement could not be prepared.\n";
-    die(print_r(sqlsrv_errors(), true));
-}
-
-sqlsrv_execute($stmt);
+function insertOneRow($conn, $tableName)
+{
+    $result = null;
+    if (AE\isColEncrypted()) {
+        $data = array("Field2" => "This is field 2.",
+                      "Field3" => array("010203", null, SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR), SQLSRV_SQLTYPE_IMAGE),
+                      "Field4" => "This is field 4.",
+                      "Field5" => array("040506", null, SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR), SQLSRV_SQLTYPE_VARBINARY('max')),
```

Comment:
why do we need to explicitly tell the driver what the PHPTYPE and SQLTYPE are?

Reply:
To bind binary types without explicitly defining both the PHPTYPE and SQLTYPE the binding will fail with different ODBC errors.

Reply:
If the hex value is used, will get `Restricted data type attribute violation` error, even with the correct SQL type. The PHP type SQLSRV_PHPTYPE_INT didn't work either.

Reply:
If I use only a character string as input, will get this ODBC error `Implicit conversion from data type varchar to varbinary(max) is not allowed. Use the CONVERT function to run this query.`

Reply:
This is a valid alternative to using a character string as input: 

array(hex2bin(0x040506), null, SQLSRV_PHPTYPE_STREAM(SQLSRV_ENC_BINARY) ),